src/ZoneServer/Scripting/AI/AiScript.cs

```diff
@@ -328,6 +333,10 @@ protected bool IsHating(ICombatEntity entity)
 			if (!_hateLevels.TryGetValue(entity.Handle, out var hate))
 				return false;
 
+			// Cloaking blocks hate while it's active
```

Comment:
In addition to the current code that disables accumulating more hate, you added a check for excluding cloaked enemies from gaining hate, as well as skipping them among the most hated entities. Effectively this means hate no longer increases and the monsters won't consider you a target for the duration of cloaking, but they will remember and target you again immediately once your cloaking status disappears. Is that the intended behavior or should the hate be cleared?

Reply:
It is indeed the intended behavior for the hate to remain and for the monster to remain hating you the moment you lose cloak, which could be almost immediately since it's disabled on attack.